modules/redisearch.ts

```diff
@@ -111,84 +116,93 @@ export class Redisearch extends Module {
                 args.push('WITHPAYLOADS')
             if(parameters.withSortKeys === true)
                 args.push('WITHSORTKEYS')
-            if(parameters.filter !== undefined)
-                args = args.concat(['FILTER', parameters.filter.field, parameters.filter.min.toString(), parameters.filter.max.toString()])
+            if(parameters.filter !== undefined) {
+                for (const filterItem of parameters.filter) {
+                    args = args.concat(['FILTER', filterItem.field, `${filterItem.min}`, `${filterItem.max}`])
+                }
+            }
             if(parameters.geoFilter !== undefined)
                 args = args.concat([
                     'GEOFILTER',
                     parameters.geoFilter.field,
-                    parameters.geoFilter.lon.toString(),
-                    parameters.geoFilter.lat.toString(),
-                    parameters.geoFilter.radius.toString(),
+                    `${parameters.geoFilter.lon}`,
+                    `${parameters.geoFilter.lat}`,
+                    `${parameters.geoFilter.radius}`,
                     parameters.geoFilter.measurement
                 ])
             if(parameters.inKeys !== undefined)
-                args = args.concat(['INKEYS', parameters.inKeys.num.toString(), parameters.inKeys.field])
+                args = args.concat(['INKEYS', `${parameters.inKeys.length}`].concat(parameters.inKeys.map(inKeysItem => `${inKeysItem}`)))
             if(parameters.inFields !== undefined)
-                args = args.concat(['INFIELDS', parameters.inFields.num.toString(), parameters.inFields.field])
+                args = args.concat(['INFIELDS', `${parameters.inFields.length}`].concat(parameters.inFields.map(inFieldItem => `${inFieldItem}`)))
             if(parameters.return !== undefined) {
-                args.push('RETURN');
-                if(parameters.return.num)
-                    args.push(parameters.return.num.toString());
-                if(parameters.return.fields)    
-                    parameters.return.fields.forEach(field => {
-                        args.push(field.name)
-                        if(field.as)
-                            args = args.concat(['AS', field.as])
-                    })
+                args.push('RETURN')
+                let itemCount = 0
+                let tempList = []
+                for (const returnItem of parameters.return) {
+                    if(typeof returnItem === "string") {
```

Comment:
It will never be a string 🤔 
The only thing you might not have is the as key.

Reply:
I think this is IF is pretty redundant, don't see the reason it's not okay to keep it as it. What are you achieving? 🤔